groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -0,0 +1,3548 @@
+            if (!valueless_by_exception()) {
+                BloombergLP::bslstl::Variant_MoveAssignVisitor<variant>
+                moveAssign(BSLS_UTIL_ADDRESSOF(lvalue));
+                BSLSTL_VARIANT_VISITID(void, variant, moveAssign, lvalue);
+            }
+        }
+        else {
+            reset();
+            if (!lvalue.valueless_by_exception()) {
+                BloombergLP::bslstl::Variant_MoveConstructVisitor<variant>
+                    moveConstructor(this, AllocBase::mechanism());
+                BSLSTL_VARIANT_VISITID(void, variant, moveConstructor, lvalue);
+            }
+            d_type = lvalue.d_type;
+        }
+    }
+    return *this;
+}
+#ifdef BSL_VARIANT_FULL_IMPLEMENTATION
+template <class HEAD, class... TAIL>
+template <class TYPE>
+inline
+typename bsl::enable_if<
+    BloombergLP::bslstl::Variant_AssignsFromType<variant<HEAD, TAIL...>,
+                                                 TYPE>::value,
+    variant<HEAD, TAIL...>&>::type
+variant<HEAD, TAIL...>::operator=(TYPE&& rhs)
+{
+    const size_t altIndex = BSLSTL_VARIANT_CONVERT_INDEX_OF(TYPE, variant);
+
+    if (index() == altIndex) {
+        bsl::get<altIndex>(*this) = std::forward<TYPE>(rhs);
+    }
+    else if (altIndex != bsl::variant_npos) {
+        emplace<altIndex>(std::forward<TYPE>(rhs));
+    }
+
+    return *this;
+}
+#else  //  BSL_VARIANT_FULL_IMPLEMENTATION
+template <class HEAD, class... TAIL>
+template <class TYPE>
+inline
+typename bsl::enable_if<
+    BloombergLP::bslstl::Variant_AssignsFromType<variant<HEAD, TAIL...>,
+                                                 TYPE>::value,
+    variant<HEAD, TAIL...>&>::type
+variant<HEAD, TAIL...>::operator=(BSLS_COMPILERFEATURES_FORWARD_REF(TYPE) rhs)
+{
+    typedef typename MoveUtil::RemoveReference<TYPE>::type ALT_TYPE;
+    const size_t altIndex = BSLSTL_VARIANT_CONVERT_INDEX_OF(ALT_TYPE, variant);
+
+    if (index() == altIndex) {
+        bsl::get<altIndex>(*this) = BSLS_COMPILERFEATURES_FORWARD(TYPE, rhs);
+    }
+    else if (altIndex != bsl::variant_npos) {
+        emplace<altIndex>(BSLS_COMPILERFEATURES_FORWARD(TYPE, rhs));
+    }
+
+    return *this;
+}
+
+#endif  // BSL_VARIANT_FULL_IMPLEMENTATION
+template <class HEAD, class... TAIL>
+inline
+void variant<HEAD, TAIL...>::swap(variant& other)
+{
+    if (!valueless_by_exception()) {
+        if (index() == other.index()) {
+            BloombergLP::bslstl::Variant_SwapVisitor<variant> swapper(this);
+            BSLSTL_VARIANT_VISITID(void, variant, swapper, other);
+        }
+        else {
+            variant tmpThis(MoveUtil::move(*this));
+            this->reset();
+            if (!other.valueless_by_exception()) {
+                BloombergLP::bslstl::Variant_MoveConstructVisitor<variant>
+                    moveConstructor(this, this->mechanism());
+                BSLSTL_VARIANT_VISITID(void, variant, moveConstructor, other);
+                this->d_type = other.d_type;
```

Comment:
This is now redundant; `Variant_MoveConstructVisitor` does it already through calling `emplace`. Ditto in two places below

Reply:
removed